shell/executables_test.ts

```diff
@@ -0,0 +1,25 @@
+import { assert, assertRejects } from '../testing_deps.ts';
+import { makeFileExecutable } from './executables.ts';
+import { $ } from './run.ts';
+
+Deno.test('new file becomes executable on unix/mac', async () => {
+  if (Deno.build.os === 'windows') {
+    return;
+  }
+
+  const filename = await Deno.makeTempFile();
+  Deno.writeTextFile(filename, 'echo "looks good"');
+  assertRejects(() => $([filename]), 'File execution succeeded before permission flag was set');
+
+  await makeFileExecutable(filename);
+  assert(await $([filename]), 'File execution failed after permission flag was set');
+});
+
+Deno.test('does not blow up on windows', async () => {
+  if (Deno.build.os !== 'windows') {
+    return;
+  }
+
+  const filename = await Deno.makeTempFile();
+  await makeFileExecutable(filename);
+});
```

Comment:
You are explicitly not testing the stated intention of the test.

```suggestion
Deno.test('does not blow up on windows', async () => {
  const filename = await Deno.makeTempFile();
  await makeFileExecutable(filename);
});
```

Reply:
This test was only meant to be applicable on windows, so if `Deno.build.os` has any other value, the test immediately exits. As written, it would still pass on linux and mac, but only because of the fact that it's not explicitly testing the terminal state.

The test above is meant to run on linux and mac, where the terminal state *is* being checked.

I reformatted these to use Deno's `ignore` option to make the checks instead of it being within the test itself. Let me know if it's unclear or I misunderstood what you're asking for.

Reply:
I apologize, I read the `if` statement incorrectly. This is totally fine. I do like the use of the ignore option though!